src/System.Device.Gpio/System/Device/Pwm/PwmChannel.cs

```diff
@@ -100,4 +100,10 @@ private static bool IsBeagleBoneKernel()
             return false;
         }
     }
+
+    /// <summary>
+    /// Query information about a component and it's children.
+    /// </summary>
+    /// <returns>A tree of <see cref="ComponentInformation"/> instances.</returns>
+    public abstract ComponentInformation QueryComponentInformation();
```

Comment:
consider

```csharp
public virtual ComponentInformation QueryComponentInformation() => new ComponentInformation(this, this.GetType().FullName);
```

Reply:
even directly on the interface

Reply:
I removed the interface as we tried to keep the public API as small as possible.